src/rgw/rgw_auth_keystone.cc

```diff
@@ -131,7 +131,7 @@ TokenEngine::get_creds_info(const TokenEngine::token_envelope_t& token
 
   return auth_info_t {
     /* Suggested account name for the authenticated user. */
-    rgw_user(token.get_project_id()),
+    rgw_user(token.get_user_name(), token.get_project_id()),
```

Comment:
this `rgw_user acct_user` field is used for several other things, so i worry about unintended consequences here:

* it's used [to check for ACL permissions](https://github.com/ceph/ceph/blob/06b3d308/src/rgw/rgw_auth.cc#L521-L527), so this could break existing ACLs

* lots of 'implicit tenant' logic depends on this tenant name being empty - for example, [this interaction with ACL checks](https://github.com/ceph/ceph/blob/06b3d308/src/rgw/rgw_auth.cc#L529-L536)

* it's the key for the 'shadow user' created by [RemoteApplier::create_account()](https://github.com/ceph/ceph/blob/06b3d308/src/rgw/rgw_auth.cc#L636-L639) and read by [RemoteApplier::load_acct_info()](https://github.com/ceph/ceph/blob/06b3d308/src/rgw/rgw_auth.cc#L676). so while rgw currently only creates one 'shadow user' per keystone project, this will now create one for each keystone user and orphan the existing per-project one

---

if we're only trying to change the ops log, the `Identity::write_ops_log_entry()` interface added in https://github.com/ceph/ceph/pull/44052 is probably a better way to do that. we could either add a 'keystone user' field to `RemoteApplier` for `RemoteApplier::write_ops_log_entry()`, or derive a `KeystoneApplier` that handles it for `TokenEngine`/`EC2Engine` specifically

Reply:
I like the "keystone user" member idea for being simple--because yes, this is just to output for the log (or similar uses) and we sure don't want to break any of those use cases